src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -389,89 +389,126 @@ trait AppDefinitionFormats {
+            version = extraFields.version
           )
         }
       }
   }
 
-  implicit lazy val AppDefinitionWrites: Writes[AppDefinition] = {
+  implicit lazy val V2AppDefinitionWrites: Writes[V2AppDefinition] = {
     implicit val durationWrites = Writes[FiniteDuration] { d =>
       JsNumber(d.toSeconds)
     }
 
-    Writes[AppDefinition] { app =>
-      Json.obj(
-        "id" -> app.id.toString,
-        "cmd" -> app.cmd,
-        "args" -> app.args,
-        "user" -> app.user,
-        "env" -> app.env,
-        "instances" -> app.instances,
-        "cpus" -> app.cpus,
-        "mem" -> app.mem,
-        "disk" -> app.disk,
-        "executor" -> app.executor,
-        "constraints" -> app.constraints,
-        "uris" -> app.uris,
-        "storeUrls" -> app.storeUrls,
-        // the ports field was written incorrectly in old code if a container was specified
-        // it should contain the service ports
-        "ports" -> app.servicePorts,
-        "requirePorts" -> app.requirePorts,
-        "backoffSeconds" -> app.backoff,
-        "backoffFactor" -> app.backoffFactor,
-        "maxLaunchDelaySeconds" -> app.maxLaunchDelay,
-        "container" -> app.container,
-        "healthChecks" -> app.healthChecks,
-        "dependencies" -> app.dependencies,
-        "upgradeStrategy" -> app.upgradeStrategy,
-        "labels" -> app.labels,
-        "acceptedResourceRoles" -> app.acceptedResourceRoles,
-        "version" -> app.version
-      )
+    Writes[V2AppDefinition] { app =>
+      try {
+        Json.obj(
+          "id" -> app.id.toString,
+          "cmd" -> app.cmd,
+          "args" -> app.args,
+          "user" -> app.user,
+          "env" -> app.env,
+          "instances" -> app.instances,
+          "cpus" -> app.cpus,
+          "mem" -> app.mem,
+          "disk" -> app.disk,
+          "executor" -> app.executor,
+          "constraints" -> app.constraints,
+          "uris" -> app.uris,
+          "storeUrls" -> app.storeUrls,
+          // the ports field was written incorrectly in old code if a container was specified
+          // it should contain the service ports
+          "ports" -> app.toAppDefinition.servicePorts,
+          "requirePorts" -> app.requirePorts,
+          "backoffSeconds" -> app.backoff,
+          "backoffFactor" -> app.backoffFactor,
+          "maxLaunchDelaySeconds" -> app.maxLaunchDelay,
+          "container" -> app.container,
+          "healthChecks" -> app.healthChecks,
+          "dependencies" -> app.dependencies,
+          "upgradeStrategy" -> app.upgradeStrategy,
+          "labels" -> app.labels,
+          "acceptedResourceRoles" -> app.acceptedResourceRoles,
+          "version" -> app.version
+        )
+      }
+      catch {
+        case NonFatal(e) =>
+          throw e
+      }
     }
   }
 
-  implicit lazy val AppUpdateReads: Reads[AppUpdate] = {
+  implicit val WithTaskCountsAndDeploymentsWrites: Writes[WithTaskCountsAndDeployments] =
```

Comment:
the vals above are all lazy to prevent problems if you access unitialized vals in your Format indirectly. I like that because it is easy to make a mistake otherwise.